Sources/Core/Extensions/Foundation/String+Povio.swift

```diff
@@ -37,4 +37,14 @@ public extension String {
     let emailTest = NSPredicate(format: "SELF MATCHES %@", emailRegEx)
     return emailTest.evaluate(with: self)
   }
+  
+  /// Returns first n characters from string
+  func first(n characters: UInt) -> String {
+    .init(prefix(min(count, Int(characters))))
+  }
+  
+  /// Returns last n characters from string
+  func last(n characters: UInt) -> String {
+    .init(suffix(min(count, Int(characters))))
+  }
```

Comment:
These two methods feel superfluous as you can just use `prefix` / `suffix` directly. What value do you think they provide?

Reply:
> Prefix/suffix allows you to put negative values inside

That's a valid point, actually! It's one of the weird things about swift and integral types. For some reason, swift prefers `IntX` over `UIntX` even when the latter makes more sense (for instance, `collection.count`).

If we decide to use such safe versions, I would suggest a different naming, smth like `safePrefix(_ maxLength: UInt)` and `safeSuffix(_ maxLength: UInt)`, which tells the meaning clearly. Second, I would return a view into the string instead of a new copy. That increases the performance when a copy is not needed and leaves that option to the user:

```swift
 /// Returns first n characters from string
  func prefix(_ maxLength: UInt) -> Substring {
    prefix(Int(maxLength))
  }

  /// Returns last n characters from string
   func suffix(_ maxLength: UInt) -> Substring {
    suffix(Int(maxLength))
  }
```